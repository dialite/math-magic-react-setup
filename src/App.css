#calculator-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.main-container {
  width: 100%;
  height: 800px;
  margin-top: 40px;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.calculation-field {
  width: 300px;
  margin-top: 20%;
}

.answer {
  width: 287px;
  height: 50px;
  padding-right: 5px;
  font-size: 20px;
  text-align: right;
  color: #fff;
  background-color: rgb(83, 90, 114);
}

.numbers-and-symbols {
  width: 100%;
  margin-top: -15px;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.size {
  width: 24.3385%;
  height: 45px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  border: 1px solid grey;
  background-color: rgb(222, 217, 217);
}

.size:hover {
  background-color: #fff;
}

.zero {
  width: 48.7%;
  height: 45px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  border: 1px solid grey;
  background-color: rgb(222, 217, 217);
}

.zero:hover {
  background-color: #fff;
}

.orabg-bg {
  background-color: rgb(252, 117, 14);
}

.HomeContainer {
  width: 100%;
  height: 400px;
}

.Header {
  width: 100%;
  height: 80px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  border: 1px solid black;
  border-radius: 8px;
}

.navigation-container {
  display: flex;
  flex-direction: row;
  list-style: none;
  gap: 20px;
  font-size: 20px;
}

.quoteContainer {
  width: 100%;
  height: 600px;
}

.quote {
  width: 100%;
  height: 600px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: larger;
}
