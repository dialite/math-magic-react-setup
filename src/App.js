import React from 'react';
import { NavLink } from 'react-router-dom';
import Calculator from './pages/calculator';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="App">
        <Calculator />
        <NavLink />
      </div>
    );
  }
}

export default App;
