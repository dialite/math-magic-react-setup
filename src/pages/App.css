@import url("https://fonts.googleapis.com/css2?family=Comfortaa:wght@300;400&display=swap");

.calculatorSection {
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin: 5rem auto;
}

.calculatorSection h1 {
  font-size: 3rem;
  font-weight: 600;
  font-style: normal;
  border-bottom: 2px solid #f5913e;
  margin-bottom: 500px;
  color: #858694;
}

.calc {
  width: 40%;
  font-size: 20px;
}

.navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #858694;
  padding: 15px 10px;
  font-weight: 600;
}

.navbar ul {
  list-style: none;
  display: flex;
}

.navbar ul > li {
  padding: 5px 10px;
  font-size: 20px;
  border-right: 1px solid white;
}

.navbar ul > li:last-child {
  border-right: none;
}

.navbar a {
  color: #f5913e;
  text-decoration: none;
  font-weight: 600;
  letter-spacing: 2px;
}

.title {
  color: #f5913e;
}

.container {
  text-align: center;
  margin: 5rem auto;
  min-height: calc(100vh - 65px);
}

.container h1 {
  font-weight: 800;
  letter-spacing: 1px;
  font-style: normal;
  font-size: 3rem;
  color: #858694;
}

.container p {
  margin: 3rem 10rem;
  letter-spacing: 1px;
  border: 1px solid #f5913e;
  padding: 2rem;
  border-radius: 8px;
  line-height: 1.3rem;
  color: #646466;
}

.container h3 {
  background: rgb(240, 239, 243);
  color: #858694;
  border: 1px solid #f5913e;
  padding: 1.5rem;
  margin: 12rem 10rem;
  border-radius: 8px;
}
