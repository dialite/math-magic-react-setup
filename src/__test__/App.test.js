import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import App from '../App';

describe('Test Router behaviour', () => {
  it('When user presses Quote in navbar it will render Quote page', () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    const quoteLink = screen.getByText('Quote');
    fireEvent.click(quoteLink);
    const main = container.querySelector('main');
    expect(main).toMatchSnapshot();
  });

  it('When user presses Home in navbar it will render Home page', () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    const homeLink = screen.getByText('Home');
    fireEvent.click(homeLink);
    const main = container.querySelector('main');
    expect(main).toMatchSnapshot();
  });

  it('When user presses Calculator in navbar it will render Calculator page', () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    const calculatorLink = screen.getByText('Calculator');
    fireEvent.click(calculatorLink);
    const main = container.querySelector('main');
    expect(main).toMatchSnapshot();
  });
});
