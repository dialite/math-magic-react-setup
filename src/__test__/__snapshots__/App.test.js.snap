// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Router behaviour When user presses Calculator in navbar it will render Calculator page 1`] = `
<main>
  <div
    class="calculator-container"
  >
    <h2>
      Lets do some math!
    </h2>
    <div
      class="calculator"
    >
      <div
        class="calculator-screen gray"
        data-testid="calculator-screen-id"
      >
        0
      </div>
      <button
        class="operator divide orange"
        type="button"
        value="รท"
      >
        รท
      </button>
      <button
        class="operator multiply orange"
        type="button"
        value="x"
      >
        ร
      </button>
      <button
        class="operator substract orange"
        type="button"
        value="-"
      >
        -
      </button>
      <button
        class="operator add orange"
        type="button"
        value="+"
      >
        +
      </button>
      <button
        class="equal-sign orange"
        type="button"
        value="="
      >
        =
      </button>
      <button
        class="all-clear"
        type="button"
        value="AC"
      >
        AC
      </button>
      <button
        class="invert-sign"
        type="button"
        value="+/-"
      >
        +/-
      </button>
      <button
        class="percentage"
        type="button"
        value="%"
      >
        %
      </button>
      <button
        class="number-key zero"
        type="button"
        value="0"
      >
        0
      </button>
      <button
        class="number-key one"
        type="button"
        value="1"
      >
        1
      </button>
      <button
        class="number-key two"
        type="button"
        value="2"
      >
        2
      </button>
      <button
        class="number-key three"
        type="button"
        value="3"
      >
        3
      </button>
      <button
        class="number-key four"
        type="button"
        value="4"
      >
        4
      </button>
      <button
        class="number-key five"
        type="button"
        value="5"
      >
        5
      </button>
      <button
        class="number-key six"
        type="button"
        value="6"
      >
        6
      </button>
      <button
        class="number-key seven"
        type="button"
        value="7"
      >
        7
      </button>
      <button
        class="number-key eight"
        type="button"
        value="8"
      >
        8
      </button>
      <button
        class="number-key nine"
        type="button"
        value="9"
      >
        9
      </button>
      <button
        class="decimal"
        type="button"
        value="."
      >
        .
      </button>
    </div>
  </div>
</main>
`;

exports[`Test Router behaviour When user presses Home in navbar it will render Home page 1`] = `
<main>
  <h2>
    Welcome to our homepage!
  </h2>
  <p>
    Lorem ipsum odor amet, consectetuer adipiscing elit. Hac luctus mi et neque non interdum nisi dapibus. Natoque vestibulum penatibus consectetur purus neque nullam. Ante sodales fermentum varius, sodales curabitur fames eros. Habitant rutrum tincidunt at luctus nam cubilia. Himenaeos facilisi suspendisse libero ac massa viverra potenti condimentum. Habitasse non hac lacinia; vulputate eget amet libero placerat. Facilisi gravida dapibus elementum arcu justo lacus felis.
  </p>
  <p>
    Venenatis dictum odio auctor consectetur cursus tristique habitasse feugiat. In nisl duis congue rhoncus egestas congue a. Faucibus aliquet fames nascetur orci placerat vehicula nunc. Mollis mauris class fames malesuada; molestie curabitur senectus. Tempor etiam cursus diam dolor mattis convallis himenaeos. Tortor varius nullam malesuada primis lacinia nascetur proin aliquam porttitor. Vel dui pretium ullamcorper pellentesque nisi per montes vehicula.
  </p>
</main>
`;

exports[`Test Router behaviour When user presses Quote in navbar it will render Quote page 1`] = `
<main>
  <div
    class="quote-container"
  >
    <p>
      Mathematics is not about numbers, equations, computations, or algorithms; it is about understanding. -William Paul Thurston
    </p>
  </div>
</main>
`;
